### DISCUSSION TITLE
"nohoist" behavior

### DISCUSSION THREAD

**Discussion by wu-json** [pnpm/pnpm#5222] [Q&A]
Is there a way to specify dependencies in a single project that should not be hoisted at all (similar to `nohoist` in a yarn workspace)? I would like to prevent my pnpm projects in my pnpm monorepo from hoisting specific react-native-related packages.

For context, I have tried using `node-linker=hoisted` with no luck. I have also tried setting `hoist=false` and setting `hoist-pattern` to exclude the react native packages of interest and it didn't seem to work. Any help on this would be great! Thanks.

**Comment by emosheeep** [comment]
Similar to https://github.com/pnpm/pnpm/discussions/5439. You can use symlink-dir to hoist dependency you want manually.

**Comment by zkochan** [comment]
Since v7.12, you can use exclude in `hoist-pattern` and `public-hoist-pattern`. For instance:

```
public-hoist-pattern[]='*types*'
public-hoist-pattern[]='!@types/react'

hoist-pattern[]='*eslint*'
hoist-pattern[]='!*eslint-plugin*'
```
